File: izinga-ordermanager/src/main/java/io/curiousoft/izinga/ordermanagement/service/ProfileService.java
package io.curiousoft.izinga.ordermanagement.service;

import io.curiousoft.izinga.commons.model.Profile;

import java.util.List;

public interface ProfileService<U extends Profile> {

    U create(U profile) throws Exception;

    U update(String profileId, U profile) throws Exception;

    void delete(String id);

    U find(String profileId);

    List<U> findAll();


}

File: izinga-usermanagement/src/main/java/io/curiousoft/izinga/usermanagement/users/ProfileService.kt
package io.curiousoft.izinga.usermanagement.users

import io.curiousoft.izinga.commons.model.Profile

interface ProfileService<U : Profile> {
    @Throws(Exception::class)
    fun create(profile: U): U

    @Throws(Exception::class)
    fun update(profileId: String, profile: U): U
    fun delete(id: String)
    fun find(profileId: String): U
    fun findAll(): List<U>
}